docs/types.rst

```diff
@@ -516,7 +516,8 @@ Every complex type, i.e. *arrays* and *structs*, has an additional
 annotation, the "data location", about whether it is stored in memory or in storage. Depending on the
 context, there is always a default, but it can be overridden by appending
 either ``storage`` or ``memory`` to the type. The default for function parameters (including return parameters) is ``memory``, the default for local variables is ``storage`` and the location is forced
-to ``storage`` for state variables (obviously).
+to ``storage`` for state variables (obviously). As of version 0.5.0, data locations for function parameters and function return parameters no longer have a default location, and ``memory`` or ``storage``
+must be provided, with the exception being parameters (not return parameters) for external functions which are talked about below.
 
 There is also a third data location, ``calldata``, which is a non-modifiable,
```

Comment:
We now have `calldata` as a keyword, so we can tell people that they have to specify `calldata` for external functions and that this is the only valid data location, right? It should even be possible to use `calldata` for local variables of reference type - but not sure if that is implemented yet (if not, just leave it open as to be implemented).

Reply:
Yes, that makes sense, however, do you think making `calldata` no longer optional would be a better fit for a separate PR? This already has some pretty big changes, and making `calldata` required would require quite a few test updates internally as well as externally (solc-js and open-zeppelin).